test/test_source_hdf5.py

```diff
@@ -0,0 +1,165 @@
+import pathlib
+import random
+import tempfile
+
+import numpy
+
+import large_image
+
+possible_axes = {
+    'x': [1, 10],
+    'y': [1, 10],
+    'c': [1, 100],
+    'z': [1, 100],
+    't': [1, 100],
+    'p': [1, 100],
+    'q': [1, 100],
+    's': [3, 3],
+}
+
+include_axes = {
+    'c': False,
+    'z': False,
+    't': False,
+    'p': False,
+    'q': False,
+}
+
+possible_data_ranges = [
+    [0, 1],
+    [0, 255],
+    [0, 65535],
+    [-1, 1]
+]
```

Comment:
I added this list but modified it slightly. Since the first two values are passed to the range function, the first is included and the second is excluded. Thus, I  removed the minus 1 from each of the end values